packages/core/src/chopsticks-worker.ts

```diff
@@ -0,0 +1,99 @@
+import { Blockchain } from './blockchain'
+import { allHandlers } from './rpc'
+import { defaultLogger as logger } from './logger'
+import { setStorage } from './utils'
+import { setup } from './setup'
+
+let chain: Blockchain | undefined
+
+const subscriptions = {}
+
+const providerHandlers = {
+  ...allHandlers,
+  new_block: async (context: any, _params: any, _subscriptionManager: any) => {
+    const { chain } = context
+    const block = await chain.newBlock()
+    return block
+  },
+}
+
+const subscriptionManager = {
+  subscribe: (method: string, subid: string, onCancel: () => void = () => {}) => {
+    subscriptions[subid] = onCancel
+    return (data: any) => {
+      postMessage({
+        type: 'subscribe-callback',
+        method,
+        subid,
+        result: JSON.stringify(data),
+      })
+    }
+  },
+  unsubscribe: (subid: string) => {
+    if (subscriptions[subid]) {
+      subscriptions[subid](subid) // call onCancel
+      postMessage({
+        type: 'unsubscribe-callback',
+        subid,
+      })
+    }
+  },
+}
+
+onmessage = async (e) => {
+  switch (e.data.type) {
+    case 'connect':
+      try {
+        logger.debug('[Chopsticks worker] onMessage: connect. Initializing...')
```

Comment:
no need to prefix message. should use child logger

`const logger = defaultLogger.child({ name: 'worker' })`